base/serialize.jl

```diff
@@ -102,14 +102,20 @@ function serialize_cycle(s::SerializationState, x)
     return false
 end
 
-serialize(s::SerializationState, x::Bool) = x ? writetag(s.io, TRUE_TAG) :
+function reset(s::AbstractSerializer)
```

Comment:
does this need exporting? is it the same meaning as the existing function? the existing one could also be renamed if it's similarly mutating.